site/blocks/CodeSnippet.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react"
+import classnames from "classnames"
+import ReactDOM from "react-dom"
+import ReactDOMServer from "react-dom/server.js"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { faCopy } from "@fortawesome/free-solid-svg-icons/faCopy"
+import { delay } from "../../clientUtils/Util.js"
+
+export const CodeSnippet = (props: { code: string }) => {
+    const [hasCopied, setHasCopied] = useState(false)
+
+    const copy = () => {
+        try {
+            navigator.clipboard.writeText(props.code)
+            setHasCopied(true)
+            // reset CSS animation
+            delay(() => setHasCopied(false), 500)
```

Comment:
I might be missing something here but I would favour the native API here, instead of introducing a new dependency.